src/models/document/base-document-content.ts

```diff
@@ -28,12 +28,18 @@ import {
   sharedModelFactory, UnknownSharedModel
 } from "../shared/shared-model-manager";
 import { IDocumentContentAddTileOptions, IDragTilesData, INewRowTile, INewTileOptions,
-   ITileCountsPerSection, NewRowTileArray } from "./document-content-types";
+   ITileCountsPerSection, NewRowTileArray, PartialSharedModelEntry, PartialTile } from "./document-content-types";
 import { SharedModelEntry, SharedModelEntryType, SharedModelEntrySnapshotType } from "./shared-model-entry";
 
 // Imports related to hard coding shared model duplication
-import { newCaseId } from "../data/data-set";
-import { SharedDataSetSnapshotType } from "../shared/shared-data-set";
+import {
+  getSharedDataSetSnapshotWithUpdatedIds, getUpdatedSharedDataSetIds, SharedDataSetSnapshotType,
+  UpdatedSharedDataSetIds, updateSharedDataSetSnapshotWithNewTileIds
+} from "../shared/shared-data-set";
+import { updateGeometryContentWithNewSharedModelIds } from "../tiles/geometry/geometry-content";
+import { updateTableContentWithNewSharedModelIds } from "../tiles/table/table-content";
+import { updateDataCardContentWithNewSharedModelIds } from "../../plugins/data-card/data-card-content";
+import { updateDataflowContentWithNewSharedModelIds } from "../../plugins/dataflow/model/dataflow-content";
```

Comment:
The problem with this approach is that it is basically including all of the tile code into the core of CLUE. Perhaps Webpack will be smart enough to know CLUE doesn't really need all of this code just for this one function, but I don't think it will.
So I'd expect after this change the core clue javascript file `index-[hash].js` will grow by a lot. 

This is why the plan for the first version of this was just work with untyped snapshots.
To keep the encapsulation you could probably get by with using `any` types and then moving these files out of there plugin and tiles folders and back into this folder. At least they would be still encapsulated. 

Then we can come back to this we can have the tiles register a generic `updateContentWithNewSharedModelIds` function. So that way the core CLUE code doesn't have to import the tiles like this.